test/core/surface/init_test.cc

```diff
@@ -44,55 +62,69 @@ static void test_blocking(int rounds) {
   for (i = 0; i < rounds; i++) {
     grpc_shutdown_blocking();
   }
+  EXPECT_FALSE(grpc_is_initialized());
+}
+
+TEST(Init, blocking) {
+  test_blocking(1);
+  test_blocking(2);
+  test_blocking(3);
+}
+
+TEST(Init, shutdown_with_thread) {
+  grpc_init();
+  {
+    INTERNAL_THREAD_SCOPE;
+    grpc_shutdown();
+  }
+  grpc_maybe_wait_for_async_shutdown();
+  EXPECT_FALSE(grpc_is_initialized());
 }
 
-static void test_mixed(void) {
+TEST(Init, mixed) {
   grpc_init();
   grpc_init();
   grpc_shutdown();
   grpc_init();
   grpc_shutdown();
   grpc_shutdown();
-  grpc_maybe_wait_for_async_shutdown();
+  EXPECT_FALSE(grpc_is_initialized());
 }
 
-static void plugin_init(void) { g_flag = 1; }
-static void plugin_destroy(void) { g_flag = 2; }
-
-static void test_plugin() {
-  grpc_register_plugin(plugin_init, plugin_destroy);
+TEST(Init, plugin) {
   grpc_init();
-  GPR_ASSERT(g_flag == 1);
+  EXPECT_TRUE(plugin_is_intialized());
   grpc_shutdown_blocking();
-  GPR_ASSERT(g_flag == 2);
+  EXPECT_TRUE(plugin_is_destroyed());
+  EXPECT_FALSE(grpc_is_initialized());
 }
 
-static void test_repeatedly() {
-  for (int i = 0; i < 1000; i++) {
+TEST(Init, repeatedly) {
```

Comment:
It feels like you should make repeatedly and repeatedly_blocking tests each have 2 variants of whether to be inline or with a thread scope.

Reply:
Inline is covered by test and test_blocking actually.